workspaces/homepage/plugins/dynamic-home-page/src/components/TemplateSection/TemplateSection.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+
+import {
+  CodeSnippet,
+  WarningPanel,
+  Link as BackstageLink,
+} from '@backstage/core-components';
+
+import Grid from '@mui/material/Grid';
+import Box from '@mui/material/Box';
+import Paper from '@mui/material/Paper';
+import CircularProgress from '@mui/material/CircularProgress';
+import CardContent from '@mui/material/CardContent';
+import { styled } from '@mui/material/styles';
+import Typography from '@mui/material/Typography';
+
+import TemplateCard from './TemplateCard';
+import { useTemplates } from '../../hooks/useTemplates';
+import { ViewMoreLink } from './ViewMoreLink';
+
+const StyledLink = styled(BackstageLink)(({ theme }) => ({
+  textDecoration: 'none',
+  padding: theme.spacing(1, 1.5),
+  fontSize: '16px',
+  display: 'inline-flex',
+  border: `1px solid ${theme.palette.primary.main}`,
+  borderRadius: 4,
+}));
+
+export const TemplateSection = () => {
+  const { data: templates, error, isLoading } = useTemplates();
+
+  const params = new URLSearchParams({
+    'filters[kind]': 'template',
+    limit: '20',
+  });
+  const catalogTemplatesLink = `/catalog?${params.toString()}`;
+
+  let content: React.ReactNode;
+
+  if (isLoading) {
+    content = (
+      <Box
+        sx={{
+          height: '100%',
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'center',
+        }}
+      >
+        <CircularProgress />
+      </Box>
+    );
+  } else if (!templates) {
+    content = (
+      <WarningPanel severity="error" title="Could not fetch data.">
+        <CodeSnippet
+          language="text"
+          text={error?.toString() ?? 'Unknown error'}
+        />
+      </WarningPanel>
+    );
+  } else {
+    content = (
+      <Box sx={{ padding: '16px 10px 10px 0' }}>
+        <React.Fragment>
+          <Grid container spacing={1} alignItems="stretch">
+            {templates?.items.map((item: any) => (
+              <Grid item xs={12} md={3} key={item.title}>
+                <TemplateCard
+                  link={`/create/templates/default/${item.metadata.name}`}
```

Comment:
This link works only when the template uses the default namespace. Please take care of `item.metadata.namespace`.